### PR TITLE
Z: short circuit ArrayCmpLen if addresses are equal

### PR DESCRIPTION
**Pull request by ehsankianifar**
If the addresses of two arrays are equal, those two arrays are considered equal and we do not need to actually compare those arrays. In [this](https://github.com/eclipse/omr/pull/6439) PR, we modified the instructions for x86 and there was a discussion to do the same for other architectures. On Z, it is more complicated and depending on the conditions and expected response, several methods implement array compare.
This PR targets array cmp len implementation. If the Array cmp asks for the position of the first mismatch, this implementation would be called. It returns the index of the fists mismatch or the length of the array if there is no mismatch.

### CODE REVIEW

**Comment by ehsankianifar** [comment]
Tested for JDK8 (LoZ and zOS) Axxon 63769, test level.sanity, all tests pass
JDK11 (LoZ) Build #20122, test sanity.functional sanity.system sanity.openjdk, all tests pass.

**Comment by ehsankianifar** [comment]
Instruction selection:
```
 [     0x2aa0bab1d00]	                       proc    	
 [     0x2aa0bab1e00]	                       fence   Relative [ 0x2aa0ba42580 ] BBStart <block_2> (frequency 10000)	
 [     0x2aa0bab2400]	                       LG      &GPR_0017, Parm[Parm  0<parm 0 L>] ?+0(GPR15)
 [     0x2aa0bad03b0]	                       LG      &GPR_0019, Parm[Parm  1<parm 1 L>] ?+8(GPR15)
 [     0x2aa0bad0670]	                       LG      GPR_0023, Parm[Parm  2<parm 2 J>] ?+16(GPR15)
 [     0x2aa0bad0d00]	                       LGR     GPR_0026,GPR_0023	
 [     0x2aa0bad0db0]	                       Label L0016:	# (Start of internal control flow)	
 [     0x2aa0bad0e70]	                       CLGR    &GPR_0017,&GPR_0019	
 [     0x2aa0bad0f20]	                       BRC     BH(0x8), Label L0017	
 [     0x2aa0bad0fe0]	                       LGR     GPR_0024,GPR_0023	
 [     0x2aa0bad1090]	                       LGR     GPR_0025,GPR_0023	
 [     0x2aa0bad1140]	                       CLCL    &GPR_0017,&GPR_0019	
 [     0x2aa0bad11f0]	                       SGR     GPR_0026,GPR_0024	
 [     0x2aa0bad1840]	                       assocreg
 [     0x2aa0bad12a0]	                       Label L0017:	# (End of internal control flow)	
 POST:
 {LegalEvenOfPair:&GPR_0017:R} {LegalOddOfPair:GPR_0024:R} {EvenOddPair:&GPR_0017:GPR_0024:R} {LegalEvenOfPair:&GPR_0019:R} {LegalOddOfPair:GPR_0025:R} {EvenOddPair:&GPR_0019:GPR_0025:R} {AssignAny:GPR_0023:R} {AssignAny:GPR_0026:R}
 [     0x2aa0bad1f50]	                       assocreg
 [     0x2aa0bad19a0]	                       retn    	
 POST:
 {GPR2:GPR_0026:R}
 [     0x2aa0bad2710]	                       fence   Relative [ 0x2aa0ba42584 ] BBEnd </block_2>	
 [     0x2aa0bad2cc0]	                       assocreg
```

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -11905,12 +11916,9 @@ OMR::Z::TreeEvaluator::arraycmplenEvaluator(TR::Node * node, TR::CodeGenerator *
    cg->stopUsingRegister(secondLen);
    cg->stopUsingRegister(orgLen);
 
-
    cg->decReferenceCount(elemsExpr);
    cg->decReferenceCount(firstBaseAddr);
    cg->decReferenceCount(secondBaseAddr);
-   cursor->setDependencyConditions(dependencies);
```
Comment (ehsankianifar) [inline compiler/z/codegen/OMRTreeEvaluator.cpp:11912]: dependency conditions added to the cFlowRegionEnd label

**Review by r30shah — APPROVED** [review]
LGTM, thanks for the change.

**Comment by r30shah** [comment]
Jenkins build zos,zlinux

**Comment by r30shah** [comment]
Jenkins build zos,zlinux

**Comment by r30shah** [comment]
@hzongaro  Can I request you to merge this change if you are OK with it ?

**Review by hzongaro — APPROVED** [review]
Looks good.  Thanks!

**Pull request merged**